scripts/test-path-filters.sh

```
@@ -0,0 +1,309 @@
+    
+    # Check if shared or critical files changed (affects all containers)
+    if [[ " ${triggered_filters[@]} " =~ " shared " ]] || \
+       [[ " ${triggered_filters[@]} " =~ " root-files " ]] || \
+       [[ " ${triggered_filters[@]} " =~ " workflows " ]] || \
+       [[ " ${triggered_filters[@]} " =~ " docker " ]]; then
+        containers_to_build=4
+        print_status $YELLOW "🔄 Shared/critical files changed - all containers will be built"
+    else
+        # Count individual container triggers
+        for filter in "bunkbot" "djcova" "starbunk-dnd" "covabot"; do
+            if [[ " ${triggered_filters[@]} " =~ " $filter " ]]; then
```

Shellcheck (0.10.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)